src/components/file-upload/file-upload.tsx

```diff
@@ -36,44 +44,87 @@ const styles: StyleRulesCallback<CssRules> = theme => ({
-    ({ classes, files, disabled, onDrop }: FileUploadProps & WithStyles<CssRules>) =>
-        <Dropzone className={classes.dropzone} onDrop={files => onDrop(files)} disabled={disabled}>
-            {files.length === 0 &&
-                <Grid container justify="center" alignItems="center" className={classes.container}>
-                    <Grid item component={"span"}>
-                        <Typography variant={"subheading"}>
-                            <CloudUploadIcon className={classes.uploadIcon} /> Drag and drop data or click to browse
-                    </Typography>
-                    </Grid>
-                </Grid>}
-            {files.length > 0 &&
-                <Table style={{ width: "100%" }}>
-                    <TableHead>
-                        <TableRow>
-                            <TableCell>File name</TableCell>
-                            <TableCell>File size</TableCell>
-                            <TableCell>Upload speed</TableCell>
-                            <TableCell>Upload progress</TableCell>
-                        </TableRow>
-                    </TableHead>
-                    <TableBody>
-                        {files.map(f =>
-                            <TableRow key={f.id}>
-                                <TableCell>{f.file.name}</TableCell>
-                                <TableCell>{formatFileSize(f.file.size)}</TableCell>
-                                <TableCell>{formatUploadSpeed(f.prevLoaded, f.loaded, f.prevTime, f.currentTime)}</TableCell>
-                                <TableCell>{formatProgress(f.loaded, f.total)}</TableCell>
-                            </TableRow>
-                        )}
-                    </TableBody>
-                </Table>
-            }
-        </Dropzone>
+    class extends React.Component<FileUploadProps, FileUploadState> {
+        constructor(props: FileUploadProps) {
+            super(props);
+            this.state = {
+                focused: false
+            };
+        }
+        render() {
+            const { classes, onDrop, disabled, files } = this.props;
+            return <div className={"file-upload-dropzone " + classes.dropzoneWrapper}>
+                <div className={classnames("dropzone-border-left", { "dropzone-border-left-active": this.state.focused })}/>
+                <div className={classnames("dropzone-border-right", { "dropzone-border-right-active": this.state.focused })}/>
+                <div className={classnames("dropzone-border-top", { "dropzone-border-top-active": this.state.focused })}/>
+                <div className={classnames("dropzone-border-bottom", { "dropzone-border-bottom-active": this.state.focused })}/>
+                <Dropzone className={classes.dropzone}
+                    onDrop={files => onDrop(files)}
+                    onClick={(e) => {
+                        const el = document.getElementsByClassName("file-upload-dropzone")[0];
```

Comment:
I wonder if we should do things like that. It's like we use React, and then we do something totally outside of it. Maybe ref prop usage would be a better option here.

Reply:
Yes, it's ugly but for some reason ref prop didn't work. I'll try to check that out one more time